packages/edit-navigation/src/components/name-editor/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { ToolbarGroup, ToolbarItem } from '@wordpress/components';
+import { useMenuEntity, useNavigationEditorMenu } from '../../hooks';
+
+export function NameEditor() {
+	const { menuName, menuId } = useNavigationEditorMenu();
+	const { editMenuName } = useMenuEntity( menuId );
+	const [ tmpMenuName, setTmpMenuName ] = useState( menuName );
+	useEffect( () => setTmpMenuName( menuName ), [ menuName ] );
+	return (
+		<>
+			<BlockControls>
```

Comment:
The input still seems to be  in the toolbar (`BlockControls`).

In the last PR we were discussing moving it to the block sidebar (`InspectorControls`) because of the accessibility issues with using an input in a toolbar.

Reply:
Yeah, let's try and go this route here (using the sidebar) as it seems to be the closest way to actually have menu name editing in place. 

If the [toolbar editing thing](https://github.com/WordPress/gutenberg/issues/23375) will be solved a11y-wise then it will make its way back into this screen as well, for editing links and other data. Until then we're moving past it, as [simple solutions](https://github.com/WordPress/gutenberg/pull/28614) are not a viable fix.

So inspector it is.
cc @shaunandrews for reference on decision.

Reply:
I posted some designs for the menu name input in the inspector sidebar: https://github.com/WordPress/gutenberg/issues/28864#issuecomment-780083954